lib/datadog/opentelemetry/extensions.rb

```diff
@@ -11,4 +13,7 @@ def self.extended(base)
       end
     end
   end
+
+  # Load and extend OpenTelemetry compatibility by default
+  extend OpenTelemetry::Extensions
```

Comment:
I really liked the simplification to auto_instrument.rb. Could we do the same here? It looks to me we could probably remove the module definition entirely and just leave a
`Datadog::Tracing::SpanOperation.prepend(Datadog::OpenTelemetry::Span)` at the top level.

Reply:
OTel is about to be removed anyways.

Reply:
Hmm, if it's going in 1.0, doesn't that mean it will live on for at least however long the 1.x series lives?